src/components/questionStructure/Task/variants/HandwritingTask/ViewOnlyCanvas.tsx

```diff
@@ -0,0 +1,31 @@
+import { Excalidraw } from '@excalidraw/excalidraw'
+import {
+  ExcalidrawImperativeAPI,
+  ExcalidrawInitialDataState,
+} from '@excalidraw/excalidraw/types/types'
+import { Box, Card } from '@radix-ui/themes'
+import React, { useEffect, useState } from 'react'
+
+interface ViewOnlyCanvasProps {
+  initialData: ExcalidrawInitialDataState
+}
+
+export const ViewOnlyCanvas: React.FC<ViewOnlyCanvasProps> = ({ initialData }) => {
+  const [excalidrawAPI, setExcalidrawAPI] = useState<ExcalidrawImperativeAPI | null>(null)
+
+  useEffect(() => {
+    setTimeout(() => {
+      excalidrawAPI?.scrollToContent(undefined, {
+        fitToContent: true,
+      })
+    })
+  }, [excalidrawAPI])
+
+  return initialData && initialData.elements?.length ? (
+    <Card>
+      <Box className="excalidraw-view-container excalidraw-box">
+        <Excalidraw excalidrawAPI={setExcalidrawAPI} viewModeEnabled initialData={initialData} />
+      </Box>
+    </Card>
+  ) : null
```

Comment:
I just noted this: I am generally inclined to avoid the `return null` pattern and place a condition on caller level i.e. from wherever `ViewOnlyCanvas` is used do

`{ initialData && <ViewOnlyCanvas initalData={initialData}/>}` or (depending on the possible shapes of initialData)
`{ !isEmpty(initialdata) && <ViewOnlyCanvas initalData={initialData}/>}` with `isEmpty` imported from `lodash` (default react dep)